From 2c48ebcd8b4dcce06ec782737225222175fef386 Mon Sep 17 00:00:00 2001
From: Sam Hartman <hartmans@debian.org>
Date: Fri, 1 Aug 2014 17:47:59 -0400
Subject: Quick and dirty fix to building -O3

This is a quick and dirty fix to pacify gcc which is over-concerned
about uninitialized variables at -O3.

This should allow Ubuntu to sync krb5 without  need for any ubuntu changes.

Patch-Category: debian-local
---
 src/kadmin/dbutil/dump.c     | 2 ++
 src/lib/krb5/os/sendto_kdc.c | 4 ++--
 src/tests/asn.1/trval.c      | 2 +-
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/kadmin/dbutil/dump.c b/src/kadmin/dbutil/dump.c
index bfb8577..f165cc9 100644
--- a/src/kadmin/dbutil/dump.c
+++ b/src/kadmin/dbutil/dump.c
@@ -1476,6 +1476,8 @@ load_db(int argc, char **argv)
     krb5_boolean db_locked = FALSE, temp_db_created = FALSE;
     krb5_boolean verbose = FALSE, update = FALSE, iprop_load = FALSE;
 
+    memset(&last, 0, sizeof(last));
+
     /* Parse the arguments. */
     dbname = global_params.dbname;
     exit_status = 0;
diff --git a/src/lib/krb5/os/sendto_kdc.c b/src/lib/krb5/os/sendto_kdc.c
index 3b3b438..462e157 100644
--- a/src/lib/krb5/os/sendto_kdc.c
+++ b/src/lib/krb5/os/sendto_kdc.c
@@ -339,7 +339,7 @@ cm_select_or_poll(const struct select_state *in, time_ms endtime,
     struct timeval tv;
 #endif
     krb5_error_code retval;
-    time_ms curtime, interval;
+    time_ms curtime = 0, interval;
 
     retval = get_curtime_ms(&curtime);
     if (retval != 0)
@@ -1315,7 +1315,7 @@ service_fds(krb5_context context, struct select_state *selstate,
             void *msg_handler_data, struct conn_state **winner_out)
 {
     int e, selret = 0;
-    time_ms endtime;
+    time_ms endtime = 0;
     struct conn_state *state;
 
     *winner_out = NULL;
diff --git a/src/tests/asn.1/trval.c b/src/tests/asn.1/trval.c
index e924fd8..3ea1edc 100644
--- a/src/tests/asn.1/trval.c
+++ b/src/tests/asn.1/trval.c
@@ -404,7 +404,7 @@ int do_cons(fp, enc, len, lev, rlen)
 {
     int n;
     int r = 0;
-    int rlen2;
+    int rlen2 = 0;
     int rlent;
     int save_appl;
 
